### PR TITLE
Merge #59: feat: Upgrade Spring Boot to version 3.0.0

### PR DESCRIPTION
**Pull request by brunoMarchiEncora**
# Description

Upgrade to Spring Boot 3 and optimized imports

Fixes [FSADT1-489](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT1-489)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

No new testes were needed

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

### CODE REVIEW

**Comment by MCatherine1994** [comment]
Hi @brunoMarchiEncora, I was looking for the swagger page, https://nrfc-api-65-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html, but it didn't show up, wondering maybe we need some new config for the swagger page after upgrade? The swagger link works in the nr-forest-client repo, could refer to the solution there, thanks!

**Review by mamartinezmejia — CHANGES_REQUESTED** [review]
Please my comment. Thanks!

File: pom.xml [hunk 1]
```diff
@@ -90,13 +97,302 @@
...
+					<skip>${checkstyle.skip}</skip>
+					<projectType>application</projectType>
+					<outputFormat>all</outputFormat>
+					<outputName>bom</outputName>
+				</configuration>
+			</plugin>
+
 		</plugins>
-		<finalName>nr-forest-client-api</finalName>
+		<finalName>nr-forest-client-backend</finalName>
```
Comment (paulushcgcj) [inline pom.xml:370]: keep the name as `nr-forest-client-api` please

File: src/main/java/ca/bc/gov/api/core/util/impl/CoreUtilImpl.java [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 
-import com.fasterxml.jackson.databind.DeserializationFeature;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializationFeature;
-
-import ca.bc.gov.api.core.util.CoreUtil;
-import ca.bc.gov.api.core.CoreConstant;
-
 @Component
 @Qualifier(CoreUtil.BEAN_NAME)
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/core/util/impl/CoreUtilImpl.java:18]: Remove the qualifier here as it is not required

File: src/main/java/ca/bc/gov/api/m/oracle/legacyclient/entity/ClientTypeCodeEntity.java [hunk 3]
```diff
@@ -1,18 +1,15 @@
...
-
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
-import ca.bc.gov.api.core.entity.AbstractEntity;
-import ca.bc.gov.api.core.misc.scope.ScopeConstant;
-
 @Entity
 @Table(name = "CLIENT_TYPE_CODE", schema = "THE")
 @Component(ClientTypeCodeEntity.BEAN_NAME)
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/m/oracle/legacyclient/entity/ClientTypeCodeEntity.java:15]: Remove component from the entity

File: src/main/java/ca/bc/gov/api/m/oracle/legacyclient/service/impl/LegacyClientServiceImpl.java [hunk 4]
```diff
@@ -15,6 +9,10 @@
 import ca.bc.gov.api.m.oracle.legacyclient.vo.ClientPublicFilterObjectVO;
 import ca.bc.gov.api.m.oracle.legacyclient.vo.ClientPublicFilterObjectVO.ClientTypeHelper;
 import ca.bc.gov.api.m.oracle.legacyclient.vo.ClientPublicViewVO;
+import javax.inject.Inject;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
 
 @Service(LegacyClientService.BEAN_NAME)
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/m/oracle/legacyclient/service/impl/LegacyClientServiceImpl.java:17]: Remove the bean name

**Review by paulushcgcj — CHANGES_REQUESTED** [review]
I know the objective of this PR is to just migrate to boot 3 but it would be nice to remove those inconsistencies like naming a bean, marking entities with scope and component, and the qualifier on top of some of the beans, as spring will handle most of this for you. It can also be done as part of the next PR.

**Comment by brunoMarchiEncora** [comment]
> Hi @brunoMarchiEncora, I was looking for the swagger page, https://nrfc-api-65-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html and https://nrfc-api-65-backend.apps.silver.devops.gov.bc.ca/swagger-ui.html, but it didn't show up, wondering maybe we need some new config for the swagger page after upgrade? The swagger link works in the nr-forest-client repo, could refer to the solution there, thanks!

Sure, let's keep this PR for upgrading to spring boot 3 and do this in another one, ok?

File: src/main/java/ca/bc/gov/api/m/oracle/legacyclient/entity/ClientTypeCodeEntity.java (hunk 3 above)
Comment (brunoMarchiEncora) [reply src/main/java/ca/bc/gov/api/m/oracle/legacyclient/entity/ClientTypeCodeEntity.java:15]: Done in next PR

File: src/main/java/ca/bc/gov/api/core/util/impl/CoreUtilImpl.java (hunk 2 above)
Comment (brunoMarchiEncora) [reply src/main/java/ca/bc/gov/api/core/util/impl/CoreUtilImpl.java:18]: Done in next PR

**Review by MCatherine1994 — APPROVED** [review]

**Review by paulushcgcj — APPROVED** [review]

**Review by mamartinezmejia — APPROVED** [review]

**Pull request merged**